src/bin/cargo-fixit/fixit.rs

```diff
@@ -161,10 +164,30 @@ fn run_rustfix(
             }
         }
 
-        file_map
-            .entry(file_name)
-            .or_insert_with(IndexSet::new)
-            .insert((suggestion, diagnostic.rendered));
+        if seen.contains(&package_id) {
+            trace!(
+                "rejecting package id `{}` already seen: {:?}",
+                package_id,
+                suggestion,
+            );
+            if let Some(rendered) = diagnostic.rendered {
+                errors.insert(rendered);
+            }
+            continue;
+        }
+
+        if current_package_id.as_ref().is_none_or(|p| p == &package_id) {
+            if current_package_id.is_none() {
+                *current_package_id = Some(package_id);
+            }
```

Comment:
Prefer to avoid double-checks like this.  Maybe something like:
```suggestion
        let current_package_id = current_package_id.get_or_insert_with(|| package_id);
        if current_package_id == package_id {
        if current_package_id.as_ref().is_none_or(|p| p == &package_id) {
            if current_package_id.is_none() {
                *current_package_id = Some(package_id);
            }
```
The `get_or_insert_with` could be inside the `if` except people don't generally expect to see side effects in an `if` so they would be more likely to misinterpret the code.